### ISSUE TITLE
model_aggregation.py

### ISSUE THREAD

**Issue by lianjinke** [YuYue525/MobileSAM-pytorch#2]
Hi author,
I have finetuned a vit_h SAM model, the encoder in it is frozen, I just finetuned prompt encoder and mask decoder. May I use your model_aggregation.py to replace my encoder with your light one?

Thanks.

**Comment by YuYue525** [comment]
I believe it works and feel free to try. However, the performance of the whole model might be worse due to the poor coupling between our encoder and your prompt encoder & mask decoder. Hope it helps you!

**Comment by lianjinke** [comment]
Hi @YuYue525 
When I use model_aggregation.py to aggregate your encoder to my finetuned vit_h SAM model, I changed line 12 of model_aggregation.py to the location of my own finetuned vit_h SAM model.
And run the script, it hints: size mismatch for image_encoder.neck.0.weight: copying a param with shape torch.Size([256, 1280, 1, 1]) from checkpoint, the shape in current model is torch.Size([256, 320, 1, 1]).
I can also change "vit_t" to "vit_h" in line 11 of model_aggregation.py and the message is similar.
Any suggestion?

**Comment by YuYue525** [comment]
> Hi @YuYue525 When I use model_aggregation.py to aggregate your encoder to my finetuned vit_h SAM model, I changed line 12 of model_aggregation.py to the location of my own finetuned vit_h SAM model. And run the script, it hints: size mismatch for image_encoder.neck.0.weight: copying a param with shape torch.Size([256, 1280, 1, 1]) from checkpoint, the shape in current model is torch.Size([256, 320, 1, 1]). I can also change "vit_t" to "vit_h" in line 11 of model_aggregation.py and the message is similar. Any suggestion?

Oh I see, you want to replace the ViT-h encoder with my ViT-t encoder, right? The script model_aggregation.py create a MobileSAM architecture and load the SAM prompt encoder & mask decoder weights, and then load the image encoder weights. 

So, if you want to replace the MobileSAM's prompt encoder and mask decoder weights, you can create a MobileSAM architecture and load your prompt encoder & mask decoder weights, and load the image encoder weights:


[pseudocode]
mobile_sam = sam_model_registry["vit_t"](checkpoint=<[mobilesam weights with my encoder](https://drive.google.com/file/d/1CitKcwSKIiYEUyM6_BxUlEABRpxVGZPf/view?usp=drive_link)>)
mobile_sam.prompt_encoder.load_state_dict(torch.load(<your prompt encoder weight path>))
mobile_sam.mask_decoder.load_state_dict(torch.load(<your prompt encoder weight path>))

**Comment by lianjinke** [comment]
Yes, I am also coding like this way, thanks brother. :-)
